File: app/src/main/java/org/schabi/newpipe/AcraReportSenderFactory.java
package org.schabi.newpipe;

import android.content.Context;

import org.acra.config.ACRAConfiguration;
import org.acra.sender.ReportSender;
import org.acra.sender.ReportSenderFactory;

/**
 * Created by the-scrabi on 13.09.16.
 */
public class AcraReportSenderFactory implements ReportSenderFactory {
    public ReportSender create(Context context, ACRAConfiguration config) {
        return new AcraReportSender();
    }
}

File: app/src/main/java/org/schabi/newpipe/error/AcraReportSenderFactory.kt
package org.schabi.newpipe.error

import android.content.Context
import com.google.auto.service.AutoService
import org.acra.config.CoreConfiguration
import org.acra.sender.ReportSender
import org.acra.sender.ReportSenderFactory

/**
 * Used by ACRA in [org.schabi.newpipe.App].initAcra() as the factory for report senders.
 */
@AutoService(ReportSenderFactory::class)
class AcraReportSenderFactory : ReportSenderFactory {
    override fun create(context: Context, config: CoreConfiguration): ReportSender {
        return AcraReportSender()
    }
}
